### ISSUE TITLE
Compile time error:  fatal error: 'Segment-Amplitude/SEGAmplitudeIntegrationFactory.h' file not found

### ISSUE THREAD

**Issue by victortimsit** [claimsforce-gmbh/flutter-segment#75]
Hey there,

I just migrated to `3.2.0` and upon running I get this compile time error:

```
Xcode build done.                                            7.0s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[49234]: Class AMSupportURLConnectionDelegate is implemented in both /usr/lib/libauthinstall.dylib (0x1ec789160) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x12c0682b8). One of the two will be used. Which one is undefined.
    objc[49234]: Class AMSupportURLSession is implemented in both /usr/lib/libauthinstall.dylib (0x1ec7891b0) and /System/Library/PrivateFrameworks/MobileDevice.framework/Versions/A/MobileDevice (0x12c068308). One of the two will be used. Which one is undefined.
    ** BUILD FAILED **
Xcode's output:
↳
    /Users/victortimsit/Documents/dev/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_segment-3.2.0/ios/Classes/FlutterSegmentPlugin.m:5:9: fatal error: 'Segment-Amplitude/SEGAmplitudeIntegrationFactory.h' file not found
    #import <Segment-Amplitude/SEGAmplitudeIntegrationFactory.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    leveldb-library-bclkldysxuuobdappophcvfqkmzu
    Command CompileSwift failed with a nonzero exit code
    leveldb-library-bclkldysxuuobdappophcvfqkmzu
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
Could not build the precompiled application for the device.

Error launching application on iPhone.
```

Any ideas?

thanks

**Comment by theochampion** [comment]
This is blocking, any updates?

**Comment by RizanPSTU** [comment]
Any update on this ?

**Comment by vcanuel** [comment]
Hi,

This error seems related to https://github.com/claimsforce-gmbh/flutter-segment/pull/72

Apparently the SEGAmplitudeIntegrationFactory.h is still referenced in `Segment_Amplitude` module.

**Comment by MaiKaY** [comment]
This import makes trouble since the release of this library. Checking the [history of the file](https://github.com/guilhermeoalima/flutter-segment/commits/87c54792ebdf56adad87c05f781e832ec48cc720/ios/Classes/FlutterSegmentPlugin.m) feels like ping pong.

@victortimsit @RizanPSTU @vcanuel @theochampion can someone please support here to finally fix it?

**Comment by vcanuel** [comment]
I have found the root cause of this issue.

TL;DR: remove " use_frameworks!" from your podfile

In my Podfile I see  that I added  "use_frameworks!"  as a requirement for certain librairies I use. 
For instance, this is needed for image_downloader ( see [wiki image_downloaded](https://github.com/ko2ic/image_downloader/wiki#image_downloaderimage_downloader-swifth-file-not-found-though-appdelegate-is-swift) and https://github.com/ko2ic/image_downloader/issues/12) 

By using  "use_frameworks!" we tells CocoaPods that you want to use Framework instead of static librairies . Unfortunately there is a name conflict between these two :(
 
I see 3 options : 
- find a temp fix to apply on a Podfile for those who need "use_frameworks!"
- Update the [segment iOs library](https://github.com/segmentio/analytics-ios) to the latest version (if it is fixed)
-  Create an [issue](https://github.com/segmentio/analytics-ios/issues)

**Comment by RizanPSTU** [comment]
I can't remove use_frameworks! because of Firebase

**Comment by mmc-eastwolf** [comment]
Quick fix : Replace in FlutterSegmentPlugin.m  
`#import <Segment-Amplitude/SEGAmplitudeIntegrationFactory.h>` 
to
`#import <Segment_Amplitude/SEGAmplitudeIntegrationFactory.h>`

**Comment by jseminck** [comment]
We have the same issue: https://github.com/claimsforce-gmbh/flutter-segment/issues/78

**Comment by vcanuel** [comment]
> Quick fix : Replace in FlutterSegmentPlugin.m `#import <Segment-Amplitude/SEGAmplitudeIntegrationFactory.h>` to `#import <Segment_Amplitude/SEGAmplitudeIntegrationFactory.h>`

This quick fix is what I initially proposed here https://github.com/claimsforce-gmbh/flutter-segment/pull/77 but this will fail the build for those who do not use `use_frameworks!` statement . So it's a chicken and egg problem !

The only "temp" solution is by forking this repo, do the fix and references it in your podspec via : 

```
  flutter_segment:
    git:
      url: git://github.com/XXX/XXXX.git <- your repo
      publish_to: none 
```

**Comment by smishal1** [comment]
any update on this ??

**Comment by Vukan-Markovic** [comment]
Any update on this?

**Comment by RizanPSTU** [comment]
> I still have this issue with the version 3.3.0 (I made `flutter clean`, removed Pods and podfile.lock, limited iOS with 12.1 at min). App crashes w/o a single call of `Segment.*` from the Dart side:
> 
> ```
> *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSCFConstantString stringByAppendingString:]: nil argument'
> *** First throw call stack:
> (0x1835d705c 0x19baf1f54 0x184d8f604 0x1042a7e34 0x1042a81e8 0x1042b0fb4 0x1042b08ac 0x1042b6630 0x183247c04 0x183249950 0x1832510ac 0x183251c10 0x18325c318 0x1f31151b0 0x1f3114f50)
> libc++abi: terminating with uncaught exception of type NSException
> * thread #66, queue = 'io.segment.analytics.segmentio', stop reason = signal SIGABRT
>     frame #0: 0x00000001b9eca9c4 libsystem_kernel.dylib`__pthread_kill + 8
> libsystem_kernel.dylib`__pthread_kill:
> ->  0x1b9eca9c4 <+8>:  b.lo   0x1b9eca9e4               ; <+40>
>     0x1b9eca9c8 <+12>: pacibsp
>     0x1b9eca9cc <+16>: stp    x29, x30, [sp, #-0x10]!
>     0x1b9eca9d0 <+20>: mov    x29, sp
> ```
> 
> Edit: Crash does not happen if I put back the deprecated keys into the Info.plist. So it looks like the underlying library does something under the hood even when not asked for that.

Getting this same error

**Comment by MaiKaY** [comment]
Fixed in `3.5.0` and please keep the following in mind.

> Since version `3.5.0` we are forcing all users to use `use_frameworks!` within the [`Podfile`](https://github.com/claimsforce-gmbh/flutter-segment/blob/master/example/ios/Podfile#L31) due to import issues of some 3rd party dependencies.
